private/buf/bufgen/config.go

```diff
@@ -157,12 +165,35 @@ func validateExternalConfigV1(externalConfig ExternalConfigV1, id string) error
 		return fmt.Errorf("%s: no plugins set", id)
 	}
 	for _, plugin := range externalConfig.Plugins {
-		if plugin.Name == "" {
-			return fmt.Errorf("%s: plugin name is required", id)
+		if plugin.Name == "" && plugin.Remote == "" {
+			return fmt.Errorf("%s: plugin name or remote plugin name is required", id)
+		}
+		if plugin.Name != "" && plugin.Remote != "" {
+			return fmt.Errorf("%s: only one of plugin name and remote plugin name can be set", id)
 		}
 		if plugin.Out == "" {
 			return fmt.Errorf("%s: plugin %s out is required", id, plugin.Name)
 		}
+		if plugin.Remote != "" {
+			if _, _, _, err := bufplugin.ParsePluginPath(plugin.Remote); err != nil {
+				return fmt.Errorf("%s: invalid remote plugin name: %w", id, err)
+			}
+			// We limit the amount of validation done on the client side intentionally
+			if plugin.Version == "" {
```

Comment:
I would probably have us do `buf.build/foo/bar:version` as the value for `remote`, and handle no version as being equivalent to latest

Reply:
I would like to use the format we use for plugins elsewhere and in the UI (`buf.build/owner/plugins/name`), but I like the idea of integrating and parsing the version out of the remote.